.github/workflows/integration.yml

```diff
@@ -0,0 +1,100 @@
+name: bootc integration test
+on:
+  pull_request:
+    branches: [main]
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        test_os: [fedora-41, fedora-42, fedora-43, centos-9]
+        test_runner: [ubuntu-latest, ubuntu-24.04-arm]
+
+    runs-on: ${{ matrix.test_runner }}
+
+    steps:
+      - name: Install podman for heredoc support
+        run: |
+          set -eux
+          echo 'deb [trusted=yes] https://ftp.debian.org/debian/ testing main' | sudo tee /etc/apt/sources.list.d/testing.list
+          sudo apt update
+          sudo apt install -y crun/testing podman/testing
+
+      - uses: actions/checkout@v4
+
+      - name: Build bootc and bootc image
+        env:
+          TEST_OS: ${{ matrix.test_os }}
+        run: sudo -E TEST_OS=$TEST_OS tests/build.sh
+
+      - name: Grant sudo user permission to archive files
+        run: |
+          sudo chmod 0755 /tmp/tmp-bootc-build/id_rsa
+
+      - name: Archive bootc disk image - disk.raw
+        if: matrix.test_runner == 'ubuntu-latest'
+        uses: actions/upload-artifact@v4
+        with:
+          name: PR-${{ github.event.number }}-${{ matrix.test_os }}-disk
+          path: /tmp/tmp-bootc-build/disk.raw
+          retention-days: 1
+
+      - name: Archive SSH private key - id_rsa
+        if: matrix.test_runner == 'ubuntu-latest'
+        uses: actions/upload-artifact@v4
+        with:
+          name: PR-${{ github.event.number }}-${{ matrix.test_os }}-id_rsa
+          path: /tmp/tmp-bootc-build/id_rsa
+          retention-days: 1
+
+  test:
+    needs: build
+    strategy:
+      matrix:
+        test_os: [fedora-41, fedora-42, fedora-43, centos-9]
+        tmt_plan: [test-01-readonly, test-20-local-upgrade, test-21-logically-bound-switch, test-22-logically-bound-install, test-23-install-outside-container, test-24-local-upgrade-reboot]
```

Comment:
It's going to be a maintenance trap if we have to list all tests here *and* in the tmt directory.

One thing that also seems like it will become a bit painful is having a CI context check per test. (It's not wrong, but it will get really long)

A good middle ground might be "sharding"; basically like https://nexte.st/docs/ci-features/partitioning/?h=shard#hashed-sharding where we run a group of tests per GHA runner. That would give us a good mix of parallelism and efficiency.

And crucially by doing the sharding we would avoid needing to list each test; the allocated runner discovers which tests it should run by just taking an integer from 0-N as input, and looking if the hash of the test modulo the input key is zero.